lib/util/fields/password_field.dart

```diff
@@ -1,39 +1,114 @@
+    this.validateStrictly = false,
+  }) : assert(validateStrictly == false || originalPasswordController == null);
+
+  @override
+  State<PasswordField> createState() => _PasswordFieldState();
+}
+
+class _PasswordFieldState extends State<PasswordField> {
+  bool _obscureText = true;
 
   @override
   Widget build(BuildContext context) {
     return TextFormField(
+      key: key,
       decoration: InputDecoration(
         border: const OutlineInputBorder(),
         labelText: labelText,
         hintText: hintText,
-        errorText: errorText,
         helperText: helperText,
+        helperMaxLines: 2,
+        suffixIcon: IconButton(
+          icon: Icon(_obscureText ? Icons.visibility : Icons.visibility_off),
+          onPressed: () {
+            setState(() {
+              _obscureText = !_obscureText;
+            });
+          },
+        ),
       ),
-      obscureText: true,
+      obscureText: _obscureText,
       enableSuggestions: false,
       autocorrect: false,
       keyboardType: TextInputType.visiblePassword,
-      controller: controller,
+      controller: widget.controller,
+      onChanged: (_) => setState(() {}),
       validator: validator,
     );
   }
+
+  bool get isConfirmation => widget.originalPasswordController != null;
+
+  String get labelText => isConfirmation ? S.of(context).formPasswordConfirm : S.of(context).formPassword;
+  String get hintText => isConfirmation
+      ? S.of(context).formPasswordConfirmHint
+      : widget.validateStrictly
+          ? S.of(context).formPasswordChooseHint
+          : S.of(context).formPasswordHint;
+
+  Key get key =>
+      widget.key ?? (isConfirmation ? PasswordField.passwordConfirmationFieldKey : PasswordField.passwordFieldKey);
+
+  String? Function(String?) get validator => isConfirmation
+      ? _validateConfirmation
+      : widget.validateStrictly
+          ? _validateStrictly
+          : _validateOnlyEmpty;
+
+  String? get helperText {
+    if (!widget.validateStrictly) return null;
+
+    final String text = widget.controller.text;
+    if (text.isEmpty) return null;
+
+    return _validateStrictly(text);
+  }
+
+  String? _validateConfirmation(String? value) {
+    if (value == null || value.isEmpty) {
+      return S.of(context).formPasswordConfirmValidateEmpty;
+    } else if (value != widget.originalPasswordController?.text) {
+      return S.of(context).formPasswordConfirmValidateMatch;
+    }
+    return null;
+  }
+
+  String? _validateOnlyEmpty(String? value) {
+    if (value == null || value.isEmpty) {
+      return S.of(context).formPasswordValidateEmpty;
+    }
+    return null;
+  }
+
+  String? _validateStrictly(String? value) {
+    if (value == null || value.isEmpty) {
+      return S.of(context).formPasswordValidateEmpty;
+    } else if (value.length < 8) {
+      return S.of(context).formPasswordValidateMinLength;
+    } else if (RegExp('[0-9]').hasMatch(value) == false) {
+      return S.of(context).formPasswordValidateMinLength;
```

Comment:
This should not be the right Return is it? 
use formPasswordValidateNumber

Reply:
Lol yes great catch. I had copied that from our existing pages, where it was wrong as well. How did no one ever notice that??

Reply:
That was a nifty copy-paste error from back when I added the translations. Glad you noticed!